src/process/windows/SDL_windowsprocess.c

```diff
@@ -0,0 +1,435 @@
+            case '"':
+            case '\\':
+            case ' ':
+            case '\t':
+                len += 2;
+                break;
+            default:
+                len += 1;
+                break;
+            }
+        }
+        /* space separator */
+        len += 1;
+    }
+
+    result = SDL_malloc(len);
+    if (!result) {
+        *args_out = NULL;
+        return false;
+    }
+
+    i_out = 0;
+    for (i = 0; args[i]; i++) {
+        const char *a = args[i];
+
+        if (i == 0) {
+            result[i_out++] = '"';
+        }
+        for (; *a; a++) {
+            switch (*a) {
+            case '"':
+            case '\\':
+            case ' ':
+            case '\t':
+                result[i_out++] = '\\';
+                result[i_out++] = *a;
+                break;
+            default:
+                result[i_out++] = *a;
+                break;
+            }
+        }
+        if (i == 0) {
+            result[i_out++] = '"';
+        }
+        result[i_out++] = ' ';
+    }
+    SDL_assert(i_out == len);
+    result[len - 1] = '\0';
+    *args_out = result;
+    return true;
+}
+
+static bool join_env(const char * const *env, char **environment_out)
+{
+    size_t len;
+    const char * const *var;
+    char *result;
+
+    if (!env) {
+        *environment_out = NULL;
+        return true;
+    }
+
+    len = 0;
+    for (var = env; *var; var++) {
+        len += SDL_strlen(*var) + 1;
+    }
+    result = SDL_malloc(len + 1);
+    if (!result) {
+        return false;
+    }
+
+    len = 0;
+    for (var = env; *var; var++) {
+        size_t l = SDL_strlen(*var);
+        SDL_memcpy(result + len, *var, l);
+        result[len + l] = '\0';
+        len += l + 1;
+    }
+    result[len] = '\0';
+
+    *environment_out = result;
+    return true;
+}
+
+bool SDL_SYS_CreateProcessWithProperties(SDL_Process *process, SDL_PropertiesID props)
+{
+    const char * const *args = SDL_GetPointerProperty(props, SDL_PROP_PROCESS_CREATE_ARGS_POINTER, NULL);
+    const char * const *env = SDL_GetPointerProperty(props, SDL_PROP_PROCESS_CREATE_ENVIRONMENT_POINTER, NULL);
+    bool enable_stdin = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN, false);
+    bool enable_stdout = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN, false);
+    bool enable_stderr = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN, false);
+    bool redirect_stderr = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_REDIRECT_STDERR_BOOLEAN, false);
+    char *createprocess_cmdline = NULL;
+    char *createprocess_env = NULL;
+    STARTUPINFOA startup_info;
+    DWORD creation_flags;
+    char *create_process_cwd;
+    SECURITY_ATTRIBUTES security_attributes;
+    HANDLE stdin_pipe[2] = { NULL, NULL };
+    HANDLE stdout_pipe[2] = { NULL, NULL };
+    HANDLE stderr_pipe[2] = { NULL, NULL };
+
+    // Keep the malloc() before exec() so that an OOM won't run a process at all
```

Comment:
Avoid null dereference in `join_arguments`.
```suggestion
    HANDLE stderr_pipe[2] = { NULL, NULL };

    if (!args) {
        return false;
    }

    // Keep the malloc() before exec() so that an OOM won't run a process at all
```

Reply:
args is guaranteed to be valid at this point.